src/azure-cli-core/azure/cli/core/parser.py

```diff
@@ -179,6 +183,13 @@ def enable_autocomplete(self):
         argcomplete.autocomplete(self, validator=lambda c, p: c.lower().startswith(p.lower()),
                                  default_completer=lambda _: ())
 
+    def _get_manual_recommendations(self, **kwargs):
```

Comment:
What's the design of this function? 
Based on the name of this function, I can see it will cover a lot of recommendations for a bigger scope but now it's just for JMESPath only.

Reply:
Yeah, you're right. It's designed to hold the flexibility for further possible errors and actionable suggestions.

It's expected we will have more hints for the common errors in the future.

Reply:
What's your plan on this? Any upcomming fetures are going to use this functions?

Reply:
Will first investigate other similar cases where CLI throw an error while not an actionable suggestion, and then see what we can do next here.